src/citrine/_utils/functions.py

```diff
@@ -124,3 +126,24 @@ def object_to_link_by_uid(json: dict) -> dict:
         return LinkByUID(scope, this_id).as_dict()
     else:
         return json
+
+
+def rewrite_s3_links_locally(url: str) -> str:
+    """Rewrites 'localstack' hosts to localhost for testing."""
+    parsed_url = urlparse(url)
+    if parsed_url.netloc != "localstack:4572":
+        return url
+    else:
+        return parsed_url._replace(netloc="localhost:9572").geturl()
+
+
+def write_file_locally(content, local_path: str):
+    """Take content from remote and ensure path exists."""
+    directory, filename = os.path.split(local_path)
+    if filename == "":
+        raise ValueError("A filename must be provided in the path")
+    if not os.path.isdir(directory):
+        os.makedirs(directory)
+    local_path = os.path.join(directory, filename)
```

Comment:
I think this line is unnecessary, since neither `directory` nor `filename` have been modified, hence `local_path` will not change.

Reply:
nice catch! Always good to save a line